woocommerce/woocommerce-gateway-stripe
# ECE - Set the order payment method title to the wallet used to process payments

Fixes #3469 

## Changes proposed in this Pull Request:

This PR makes the following changes:

1. Removes code added in a previous ECE PR (https://github.com/woocommerce/woocommerce-gateway-stripe/pull/3440) which set the payment method title when processing payments via the Pay for Order payment. This is no longer necessary because we're setting the payment method title in a more general way that will work for all pages. 
2. Stripe payment method objects record the type of express payment wallet is used to process the payment (see example below). When processing a payment with wallet information, set the payment method title to the corresponding wallet type. 

<details><summary>Stripe Payment Method object</summary>
<p>

```
stdClass Object
(
    [id] => pm_1Q4c4tEYkV5SNaUVCGyhKgeg
    [object] => payment_method
    [billing_details] => stdClass Object
        (
            [address] => stdClass Object
                (
                   ...
                )

            [email] => ...
            [name] => Card Holder Name
            [phone] => +1 650-555-5555
        )

    [card] => stdClass Object
        (
            [brand] => visa
            ...

            [wallet] => stdClass Object
                (
                    [dynamic_last4] => 1111
                    [google_pay] => stdClass Object
                        (
                        )

                    [type] => google_pay <------ SEE HERE
                )

        )
     ...

    [type] => card
)
```
</p>
</details>

## Testing instructions

1. Set the ECE feature flag option (`_wcstripe_feature_ece`) to `'yes'`.
2. Enable Express payment methods Google and Apple Pay.
3. Complete a purchase using Google and/or Apple Pay
4. Confirm on the order received page that the order's payment method title is set to Google or Apple Pay. 

<img width="363" alt="Screenshot 2024-09-30 at 3 35 37 pm" src="https://github.com/user-attachments/assets/7e9d28bd-a230-40bb-bdaa-b9f48adae267">

While working on this, I've also confirmed that standard card payments still set the title to the [default `Credit / Debit Card`](https://github.com/user-attachments/assets/44e344b9-b066-4e74-994f-b692e51345dd) and [if the title is customised](https://github.com/user-attachments/assets/0cb24320-2066-4121-b8c0-4d3d55cd8422), [it is still set](https://github.com/user-attachments/assets/5b3dcf9e-2db4-4622-8833-83d488119de2) for standard card payments. 


---


**Post merge**
